ppb_vector/vector2.py

```diff
@@ -1,97 +1,174 @@
 import typing
 import collections
+import functools
 from math import acos, atan2, cos, degrees, hypot, isclose, radians, sin
 from numbers import Real
 from collections.abc import Sequence
 
 __all__ = 'Vector2',
 
 
+# Vector or subclass
+VectorOrSub = typing.TypeVar('VectorOrSub', bound='Vector2')
+
+Realish = typing.Union[Real, float, int]
+
+# Anything convertable to a Vector, including lists, tuples, and dicts
 VectorLike = typing.Union[
-    'Vector2',
-    typing.List[Real],  # TODO: Length 2
-    typing.Tuple[Real, Real],
-    typing.Dict[str, Real],  # TODO: Length 2, keys 'x', 'y'
+    'Vector2',  # Or subclasses, unconnected to the VectorOrSub typevar above
+    typing.List[typing.SupportsFloat],  # TODO: Length 2
+    typing.Tuple[typing.SupportsFloat, typing.SupportsFloat],
```

Comment:
Replace `List` and `Tuple` with `typing.Sequence` ?

Reply:
The code only supports `list` and `tuple`, so ideally yes, but that's not what's implemented yet.